src/vm/gchelpers.cpp

```diff
@@ -60,6 +60,40 @@ inline gc_alloc_context* GetThreadAllocContext()
     return & GetThread()->m_alloc_context;
 }
 
+// Checks to see if the given allocation size exceeds the
+// largest object size allowed - if it does, it throws
+// an OutOfMemoryException with a message indicating that
+// the OOM was not from memory pressure but from an object
+// being too large.
+inline void CheckObjectSizeOrThrow(size_t alloc_size)
+{
```

Comment:
This may need:
```
 CONTRACTL {
        THROWS;
        GC_TRIGGERS;
    } CONTRACTL_END;
```

to prevent the static contract verifier in internal build from complaining.